cli/ballerina-launcher/src/main/java/org/ballerinalang/launcher/Main.java

```diff
@@ -80,77 +75,73 @@ public static void main(String... args) {
             ServiceLoader<BLauncherCmd> bCmds = ServiceLoader.load(BLauncherCmd.class);
             for (BLauncherCmd bCmd : bCmds) {
-                cmdParser.addCommand(bCmd.getName(), bCmd);
+                cmdParser.addSubcommand(bCmd.getName(), bCmd);
                 bCmd.setParentCmdParser(cmdParser);
             }
 
             // Build Version Command
             VersionCmd versionCmd = new VersionCmd();
-            cmdParser.addCommand(BallerinaCliCommands.VERSION, versionCmd);
+            cmdParser.addSubcommand(BallerinaCliCommands.VERSION, versionCmd);
             versionCmd.setParentCmdParser(cmdParser);
 
             EncryptCmd encryptCmd = new EncryptCmd();
-            cmdParser.addCommand(BallerinaCliCommands.ENCRYPT, encryptCmd);
+            cmdParser.addSubcommand(BallerinaCliCommands.ENCRYPT, encryptCmd);
             encryptCmd.setParentCmdParser(cmdParser);
 
-            cmdParser.setProgramName("ballerina");
-            cmdParser.parse(args);
-            String parsedCmdName = cmdParser.getParsedCommand();
+            cmdParser.setCommandName("ballerina");
 
-            // User has not specified a command. Therefore returning the main command
-            // which simply prints usage information.
-            if (parsedCmdName == null) {
-                return Optional.of(defaultCmd);
-            }
+            cmdParser.setStopAtPositional(true);
 
-            Map<String, JCommander> commanderMap = cmdParser.getCommands();
-            return Optional.of((BLauncherCmd) commanderMap.get(parsedCmdName).getObjects().get(0));
+            List<CommandLine> parsedCommands = cmdParser.parse(args);
 
-        } catch (MissingCommandException e) {
-            String errorMsg = "unknown command '" + e.getUnknownCommand() + "'";
-            throw LauncherUtils.createUsageException(errorMsg);
+            if (parsedCommands.size() < 1) {
+                return Optional.of(defaultCmd);
+            }
 
-        } catch (ParameterException e) {
+            return Optional.of(parsedCommands.get(parsedCommands.size() - 1).getCommand());
+        } catch (CommandLine.UnmatchedArgumentException e) {
+            String errorMessage = e.getMessage();
+            if (errorMessage == null) {
+                throw LauncherUtils.createUsageException("internal error occurred");
+            }
+            if (errorMessage.contains(UNMATCHED_ARGUMENT_PREFIX)) {
+                throw LauncherUtils.createUsageException("unknown command '" + getFirstUnknownArg(errorMessage)
+                                                                 + "'");
+            }
+            throw LauncherUtils.createUsageException(LauncherUtils.makeFirstLetterLowerCase(errorMessage));
+        } catch (CommandLine.ParameterException e) {
             String msg = e.getMessage();
             if (msg == null) {
                 throw LauncherUtils.createUsageException("internal error occurred");
```

Comment:
We don't have to prepend "ballerina: " when `createUsageException` is called since it is prepend-ed at https://github.com/ballerina-platform/ballerina-lang/blob/master/cli/ballerina-launcher/src/main/java/org/ballerinalang/launcher/LauncherUtils.java#L173, right?